contrib/pkg/aws_tag_deprovision/aws_tag_deprovision.go

```diff
@@ -697,9 +682,43 @@ func deleteInstanceProfile(instanceProfileID *string, iamClient *iam.IAM, logger
 	return nil
 }
 
+// deleteIAMresources will delete any IAM resources created by the installer that are not associated with a running instance
+// Currently openshift/installer creates 3 roles per cluster, 1 for master|worker|bootstrap and identified by the
+// cluster name used to install the cluster.
+func deleteIAMresources(session *session.Session, filter awsFilter, clusterName string, logger log.FieldLogger) (bool, error) {
+	logger.Debugf("Deleting IAM resources")
+	defer logger.Debugf("Exiting deleting IAM resources")
+	iamClient := iam.New(session)
+	if clusterName != "" {
+		installer_type := []string{"master", "worker", "bootstrap"}
+		for _, t := range installer_type {
+			instanceProfileName := fmt.Sprintf("%s-%s-profile", clusterName, t)
+			describeProfileInput := iam.GetInstanceProfileInput{}
+			describeProfileInput.InstanceProfileName = &instanceProfileName
+			profile, err := iamClient.GetInstanceProfile(&describeProfileInput)
+			if err != nil {
+				logger.Debugf("error listing instance-profile: %v", err)
+				return false, nil
+			}
+			// first delete any roles out of the instance profile
+			err = deleteRolesFromInstanceProfile(profile.InstanceProfile, iamClient, logger)
+			if err != nil {
+				logger.Debugf("error deleting roles from instance profile: %v", err)
+				return false, nil
+			}
+			err = deleteInstanceProfile(profile.InstanceProfile.InstanceProfileId, iamClient, logger)
+			if err != nil {
+				logger.Debugf("error deleting instance profile: %v", err)
+				return false, nil
+			}
+		}
+	}
+	return true, nil
+}
+
```

Comment:
there could be a way so we don't have to depend on the hardcoded installer_type above.. or a better name than `installer_type`..